gallery/src/components/NxTable/NxTableFilterExample.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, {useState} from 'react';
+
+import {
+  NxTable,
+  NxTableBody,
+  NxTableCell,
+  NxTableHead,
+  NxTableRow
+} from '@sonatype/react-shared-components';
+import { toLower } from 'ramda';
+
+const initialState = [
+  {name: 'Anna', country: 'USA'},
+  {name: 'Lean', country: 'France'},
+  {name: 'Louis', country: 'France'},
+  {name: 'Zach', country: 'Colombia'},
+  {name: 'Jimmy', country: 'Germany'}
+];
+
+const NxTableFilterExample = () => {
+
+  const [rows, setRows] = useState(initialState);
+  const [filter, setFilter] = useState('');
+
+  const getCountries = (list: { name: string; country: string }[]) => {
+    return list.map((state: any) => {
+      return state['country'];
+    }).sort();
+  };
+
+  const dropdownDefaultLabel = 'All countries';
+  const countries = getCountries(initialState);
+  const [optionsDropdown, setOptionsDropdown] = useState<Set<string>>(new Set(countries));
+  const [filterDropdownLabel, setFilterDropdownLabel] = useState(dropdownDefaultLabel);
+  const [filterDropdown, setFilterDropdown] = useState('');
+
+  const applyFilter = (rows: { name: string; country: string }[], name: string, country: string) => {
+    let filteredRows = rows;
+    if (name !== '') {
+      filteredRows = filteredRows.filter((row: any) => toLower(row.name).includes(toLower(name)));
+    }
+    if (country !== '' && country !== dropdownDefaultLabel) {
+      filteredRows = filteredRows.filter((row: any) => toLower(row.country).includes(toLower(country)));
+    }
+    return filteredRows;
+  };
+
+  const handleOptionsDropdown = (country: string) => {
+    if (country === '') {
+      setOptionsDropdown(new Set(countries));
+    }
+    else {
+      const dropDownOptions = [dropdownDefaultLabel].concat(countries.filter(c => c !== country));
```

Comment:
Once I've chosen a country I added 'All countries' in the dropdown options in order to set the rows to the initial state
![image](https://user-images.githubusercontent.com/57679720/95870112-6bbb5400-0d6c-11eb-87ca-73b8bc0a071c.png)